### PR TITLE
gh-94061: set a message to ConnectionResetError

### PR DESCRIPTION
**Pull request by sileht**
The asyncio ssl error handling will now set a message for
ConnectionResetError exception.

* Issue: gh-94061

### CODE REVIEW

**Comment by sileht** [comment]
@vstinner can you take a look?

File: Lib/asyncio/sslproto.py [hunk 1]
```diff
@@ -460,7 +460,8 @@ def eof_received(self):
                 logger.debug("%r received EOF", self)
 
             if self._state == SSLProtocolState.DO_HANDSHAKE:
-                self._on_handshake_complete(ConnectionResetError)
+                self._on_handshake_complete(
+                    ConnectionResetError("SSL Connection reset by peer"))
```
Comment (vstinner) [inline Lib/asyncio/sslproto.py:461]: I'm not convinced that it's always the case. SSLProtocol._do_shutdown() calls _call_eof_received().

If you want to pass a reason, I suggest adding a parameter to _call_eof_received(). And maybe add a private _eof_received() function which accepts a message. For example, eof_received() can simply call _eof_received() with a default message.

close() also starts a "shutdown" process which indirectly calls _call_eof_received().

So there are many ways to reach this line, and different reasons causing it.

Comment (sileht) [reply Lib/asyncio/sslproto.py:461]: I don't think we have other reasons of why the connection is being reset here.
`ConnectionResetError` here is only raised if the state of the connection is `SSLProtocolState.DO_HANDSHAKE`.

The `_call_eof_received()` method can be called only by `_do_shutdown()` and `_do_read()`
* `_do_shutdown()` is only called if the state is `SSLProtocolState.SHUTDOWN`.
* `_do_read()` is only called if the state is `SSLProtocolState.WRAPPED`.

So this exception can only be raised at the early stage of the SSL connection establishment.

**Comment by vstinner** [comment]
Well, it seems like I'm not the right person to review this change. @pitrou @1st1: Would you mind to have a look at this asyncio enhancement?

**Comment by sileht** [comment]
@vstinner thanks!

**Comment by sileht** [comment]
@pitrou @1st1 can I get some insight, to get that in?

**Comment by gvanrossum** [comment]
Won't fix, see issue.

**Pull request closed without merging**